src/main/java/com/google/api/codegen/transformer/ruby/RubyTestsTransformer.java

```diff
@@ -0,0 +1,252 @@
+
+import com.google.api.codegen.InterfaceView;
+import com.google.api.codegen.ResourceNameTreatment;
+import com.google.api.codegen.ServiceMessages;
+import com.google.api.codegen.config.ApiConfig;
+import com.google.api.codegen.config.FieldConfig;
+import com.google.api.codegen.config.FlatteningConfig;
+import com.google.api.codegen.config.MethodConfig;
+import com.google.api.codegen.config.ResourceNameMessageConfigs;
+import com.google.api.codegen.gapic.GapicCodePathMapper;
+import com.google.api.codegen.metacode.InitCodeContext;
+import com.google.api.codegen.metacode.InitCodeContext.InitCodeOutputType;
+import com.google.api.codegen.transformer.FileHeaderTransformer;
+import com.google.api.codegen.transformer.InitCodeTransformer;
+import com.google.api.codegen.transformer.MethodTransformerContext;
+import com.google.api.codegen.transformer.ModelToViewTransformer;
+import com.google.api.codegen.transformer.ModelTypeTable;
+import com.google.api.codegen.transformer.SurfaceNamer;
+import com.google.api.codegen.transformer.SurfaceTransformerContext;
+import com.google.api.codegen.util.Name;
+import com.google.api.codegen.util.SymbolTable;
+import com.google.api.codegen.util.ruby.RubyTypeTable;
+import com.google.api.codegen.util.testing.JavaValueProducer;
+import com.google.api.codegen.util.testing.TestValueGenerator;
+import com.google.api.codegen.viewmodel.ApiMethodType;
+import com.google.api.codegen.viewmodel.InitCodeView;
+import com.google.api.codegen.viewmodel.ViewModel;
+import com.google.api.codegen.viewmodel.testing.GapicSurfaceTestAssertView;
+import com.google.api.codegen.viewmodel.testing.GapicSurfaceTestCaseView;
+import com.google.api.codegen.viewmodel.testing.GapicSurfaceTestClassView;
+import com.google.api.codegen.viewmodel.testing.MockGrpcResponseView;
+import com.google.api.codegen.viewmodel.testing.MockServiceUsageView;
+import com.google.api.codegen.viewmodel.testing.PageStreamingResponseView;
+import com.google.api.tools.framework.model.Field;
+import com.google.api.tools.framework.model.Interface;
+import com.google.api.tools.framework.model.Method;
+import com.google.api.tools.framework.model.Model;
+import com.google.common.collect.ImmutableMap;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+
+/** A subclass of ModelToViewTransformer which translates model into API tests in Ruby. */
+public class RubyTestsTransformer implements ModelToViewTransformer {
+  // Template files
+  private static String UNIT_TEST_TEMPLATE_FILE = "ruby/test.snip";
+
+  private final FileHeaderTransformer fileHeaderTransformer = new FileHeaderTransformer();
+  private final InitCodeTransformer initCodeTransformer = new InitCodeTransformer();
+  // It looks like this isn't producing anything that's specific to Java?
+  // TODO(jcanizales): Use same value generator as sample gen!
+  private final TestValueGenerator valueGenerator = new TestValueGenerator(new JavaValueProducer());
```

Comment:
I think it's much preferable to come up with a general name that makes room for other languages, so that it doesn't have to be renamed for each language it is used for.